### DISCUSSION TITLE
Allways after restarting the computer I have to restart explorer

### DISCUSSION THREAD

**Discussion by Anixx** [valinet/ExplorerPatcher#841] [General]
Every time after restarting the computer, the tray and taskbar not working. I have to enter the preferences and choose to restart explorer.

**Comment by Anixx** [comment]
After boot I see empty taskbar and tray: 
![изображение](https://user-images.githubusercontent.com/1271736/153683903-2f3866fd-4f18-4e2a-b03e-d48f5136d06f.png)

**Reply by valinet** [reply]
Lots of things have to happen to hack the classic theme correctly. Maybe increase the "Supplementary delay at logon" in Advanced? Otherwise, yeah, restart Explorer at startup or determine what would need to change to have it startup correctly.

**Reply by Anixx** [reply]
In early versions of EP all worked well... Now this problem is persistent (writing after logging in and restarting explorer)

**Reply by valinet** [reply]
It’s best to try to identify the commit where the regression happened. Try to determine that, that’s a better starting point than repeating what you have already said 3 days ago.

**Reply by Anixx** [reply]
Rolled back to 41.16 (current release). Seems to be OK so far. I will try other builds.

**Reply by Anixx** [reply]
Also, noticed that the properties window is white in the latest build white in the current stable release 41.16 it respects the system color scheme:
![изображение](https://user-images.githubusercontent.com/1271736/154683781-b1e2445f-2ad5-4fee-b01b-6c8c1e4f562d.png)

**Reply by l0oo** [reply]
Hello
I have a similar 'classic scheme'. Didn't notice your issue with the tray and taskbar not working after restarting. Maybe smthng what is loading to tray makes such stuck? :/ 

2nd thing - I also have white background the properties window now (42.15). :( 
The system colour scheme was much more pleasant to the eye.

PS
It may be not the reason, but try to change 'Start button style: Windows 11' to Windows 10. I have W10 there + using Open-Shell Menu.

**Reply by Anixx** [reply]
@l0oo thanks, tried your advice, indeed it was win11 style for some reason.

**Reply by valinet** [reply]
The background color change is due to this: https://github.com/valinet/ExplorerPatcher/blob/master/ExplorerPatcher/GUI.c#L905

In previous versions it was set to `COLOR_MENU`, while now I have set it to `COLOR_WINDOW` which according to MSDN is "Window background. The associated foreground colors are COLOR_WINDOWTEXT and COLOR_HOTLITE". `COLOR_MENU` is "Menu background. The associated foreground color is COLOR_MENUTEXT". So `COLOR_WINDOW` looks correct to me. The change was made since the window displayed wrongly when using high contrast themes, where the same code path for rendering the window is employed, as if using classic theme. Idk, I can differentiate between the 2, but `COLOR_WINDOW` seems correct to me, in fact.

**Reply by Anixx** [reply]
COLOR_WINDOW is definitely wrong. It is the same color as the color of text background in say, notepad in Classic theme, that is by default white. On the image below I chose it to be pink:

![изображение](https://user-images.githubusercontent.com/1271736/154796649-c3e31f00-1c09-4493-bf35-a6993ce34a29.png)

What you really need is the so-called color of 3D objects:

![изображение](https://user-images.githubusercontent.com/1271736/154796703-cbee34b3-b013-4842-a0e4-3f98c71954ed.png)

**Reply by Anixx** [reply]
Actually, you need COLOR_3DFACE https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getsyscolor
It is officially intended for dialog boxes background.

**Reply by valinet** [reply]
Okay. Changed that, published a hotfix. Check for updates (pre-release).

**Reply by l0oo** [reply]
TYVM valinet! 
Works great now! :)

**Reply by Anixx** [reply]
It seems, the original issue with empty tray and taskbar has gone. Either due to the advice by @l0oo regarding the start button or because I uninstalled the Foobar player's toolbar. I will test it more.
